File: Backend/src/main/java/com/shopsphere/backend/service/OrderService.java
package com.shopsphere.backend.service;

import com.shopsphere.backend.dto.order.OrderResponse;
import com.shopsphere.backend.dto.order.PlaceOrderRequest;

import java.util.List;

public interface OrderService {

    // Customer features

    OrderResponse placeOrder(
            PlaceOrderRequest request
    );

    List<OrderResponse> getMyOrders();

    OrderResponse getOrderById(
            Long orderId
    );

    void cancelOrder(
            Long orderId
    );


    // Admin features

    List<OrderResponse> getAllOrders();

    OrderResponse getOrderByIdForAdmin(
            Long orderId
    );

    OrderResponse updateOrderStatus(
            Long orderId,
            String status
    );
}

File: shopsphere-frontend/src/services/orderService.js
import axiosInstance from "./axiosInstance";

export const placeOrder = (addressId) => {
  return axiosInstance.post(
    "/orders/place",
    {
      addressId: addressId,
    }
  );
};

export const getMyOrders = () => {
  return axiosInstance.get("/orders");
};

export const getOrderById = (orderId) => {
  return axiosInstance.get(
    `/orders/${orderId}`
  );
};

export const cancelOrder = (orderId) => {
  return axiosInstance.put(
    `/orders/${orderId}/cancel`
  );
};